### ISSUE TITLE
Assignment to scalar in forall warning/error

### ISSUE THREAD

**Issue by schweitzpgi** [flang-compiler/f18-llvm-project#1058]
```fortran
  integer :: i, n
  real :: y
  forall (i=1:n) y = i**2
end
```
where n, y, and i are all scalar.
Flang emits:
```
Warning: FORALL index variable 'i' not used on left-hand side of assignment
  forall (i=1:n) y = i**2
                 ^^^^^^^^
```
It does not emit a diagnostic about `y` being a scalar and thus trivially being a "many-to-one assignment" situation in violation of 10.2.4.4(p)1.

**Comment by klausler** [comment]
Is it still a violation if n is 1?

**Comment by schweitzpgi** [comment]
> Is it still a violation if n is 1?

I'll update the example.
